src/nvim/screen.c

```diff
@@ -4199,8 +4199,18 @@ win_line (
             || (wp->w_p_list && lcs_eol != NUL && p_extra != at_end_str)
             || (n_extra != 0 && (c_extra != NUL || *p_extra != NUL)))
         ) {
+      // Special trick to make copy/paste of wrapped lines work with
+      // xterm/screen: write the next character after the end of the
+      // line. This will work with all terminal types (regardless of the
+      // xn,am settings).
+      bool wrap = wp->w_p_wrap     // Wrapping enabled.
+        && filler_todo <= 0        // Not drawing diff filler lines.
+        && lcs_eol_one != -1       // Haven't printed the lcs_eol character.
+        && row != endrow - 1       // Not the last line being displayed.
+        && wp->w_width == Columns  // Window spans the width of the screen.
+        && !wp->w_p_rl;            // Not right-to-left.
```

Comment:
I've removed the restriction on doing this for lines ending in multibyte characters. As far as I can tell, this restriction dates back to 2000, when it was added to work around a bug in kterm (https://marc.info/?l=vim-dev&m=96549912530900&w=2). Since the TUI operates in terms of cells rather than individual bytes, this should not do anything funny for multibyte characters.